packages/datetime/src/dateRangeInput.tsx

```diff
@@ -66,6 +73,28 @@ export interface IDateRangeInputProps extends IDatePickerBaseProps, IProps {
     onChange?: (selectedRange: DateRange) => void;
 
     /**
+     * Called when the user finishes typing in a new date and the date causes an error state.
+     * If the date is invalid, `new Date(undefined)` will be returned for the corresponding
+     * boundary of the date range.
+     * If the date is out of range, the out-of-range date will be returned for the corresponding
+     * boundary of the date range (`onChange` is not called in this case).
+     */
+    onError?: (errorRange: DateRange) => void;
+
+    /**
+     * The error message to display when the date selected is out of range.
+     * @default "Out of range"
+     */
+    outOfRangeMessage?: string;
+
+    /**
+     * The error message to display when the selected dates overlap.
+     * This can only happen when typing dates in the input field.
+     * @default "Overlapping dates"
+     */
+    overlappingDatesMessage?: string;
```

Comment:
this feature confuses me. seems like unnecessary pain to reject this almost-valid state (particularly when the DRI handles this case itself when clicking). in your gif at the end of the description, i'd actually expect the DRI to just correct the dates so the earlier one is on the left.